[TZone] Add TZone annotation macros for structs

msaboff opened the pull request:
#### 71c9542da1c301de4ab7c96c541a8cbc97d6c432
<pre>
[TZone] Add TZone annotation macros for structs
<a href="https://bugs.webkit.org/show_bug.cgi?id=281514">https://bugs.webkit.org/show_bug.cgi?id=281514</a>
<a href="https://rdar.apple.com/137985417">rdar://137985417</a>

Reviewed by Yijia Huang.

Split the existing MAKE_BTZONE_MALLOCED macro into MAKE_BTZONE_MALLOCED_COMMON, that doesn&apos;t have access specifiers,
and MAKE_BTZONE_MALLOCED, which invokes MAKE_BTZONE_MALLOCED_COMMON with a public specifier before and a
private specifier after.  Created a new MAKE_STRUCT_BTZONE_MALLOCED that just invokes MAKE_BTZONE_MALLOCED_COMMON.
Both the class and struct macros include a dummy using statement to allow them to be followed by a semicolon.
Made similar changes to MAKE_BTZONE_MALLOCED_INLINE.

Added WTF_MAKE_STRUCT_TZONE_ALLOCATED and WTF_MAKE_STRUCT_TZONE_ALLOCATED_INLINE which call the new MAKE_BTZONE_* macros.

Used both of the new WTF_MAKE_STRUCT_TZONE_ALLOCATED* macros in a few files, replacing WTF_MAKE_STRUCT_FAST_ALLOCATED in
some places and added WTF_MAKE_STRUCT_TZONE_ALLOCATED in a few places where there wasn&apos;t a WTF_MAKE_STRUCT_FAST_ALLOCATED.

Note that the nested struct RemoteObjectInvocation::ReplyInfo was previously annotated using the &quot;class&quot; annotation macro.
This worked because RemoteObjectInvocation::ReplyInfo had an existing public: specifier.  For consistency,
changed RemoteObjectInvocation::ReplyInfo to use WTF_MAKE_STRUCT_TZONE_ALLOCATED instead and removed the public: specifier.

* Source/JavaScriptCore/bytecode/LazyValueProfile.h:
* Source/JavaScriptCore/bytecode/UnlinkedFunctionExecutable.cpp:
* Source/JavaScriptCore/bytecode/UnlinkedFunctionExecutable.h:
* Source/JavaScriptCore/dfg/DFGOSRExit.cpp:
* Source/JavaScriptCore/dfg/DFGOSRExit.h:
* Source/JavaScriptCore/inspector/remote/socket/RemoteInspectorSocketEndpoint.cpp:
* Source/JavaScriptCore/inspector/remote/socket/RemoteInspectorSocketEndpoint.h:
* Source/JavaScriptCore/jit/JITMathIC.h:
* Source/JavaScriptCore/jit/JITTZoneImpls.cpp:
* Source/JavaScriptCore/parser/VariableEnvironment.cpp:
* Source/JavaScriptCore/parser/VariableEnvironment.h:
* Source/WTF/wtf/TZoneMalloc.h:
* Source/WTF/wtf/TZoneMallocInlines.h:
* Source/WebCore/Modules/airplay/WebMediaSessionManager.cpp:
* Source/WebCore/PAL/pal/crypto/commoncrypto/CryptoDigestCommonCrypto.cpp:
* Source/WebCore/rendering/BidiRun.cpp:
* Source/WebCore/rendering/BidiRun.h:
* Source/WebKit/Shared/API/Cocoa/RemoteObjectInvocation.h:
* Source/bmalloc/bmalloc/TZoneHeap.h:
* Source/bmalloc/bmalloc/TZoneHeapInlines.h:

Canonical link: <a href="https://commits.webkit.org/285280@main">https://commits.webkit.org/285280@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/04e95aa4bc25cf6c2bab48e8e3e6bca5f724284f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/71904 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/51324 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/24689 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/76064 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/23113 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/74019 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/59125 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/22933 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/56752 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/15259 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/74971 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/46553 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/61945 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/37190 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/43216 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/19415 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/21458 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/65032 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/65128 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/19779 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/77744 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/71156 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/16144 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/18964 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/65020 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/16190 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/61968 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/64496 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/12667 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/6306 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/92939 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11062 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/47122 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/1906 "Built successfully") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/20480 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/48191 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/49478 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/47935 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

dhecht commented on Source/WebKit/Shared/API/Cocoa/RemoteObjectInvocation.h:
```diff
@@ -40,7 +40,7 @@ class RemoteObjectInvocation {
     struct ReplyInfo {
-        WTF_MAKE_TZONE_ALLOCATED(ReplyInfo);
+        WTF_MAKE_STRUCT_TZONE_ALLOCATED(ReplyInfo);
     public:
```
could remove this `public:` while here

msaboff replied:
Sure, I'll remove it since it isn't needed.

dhecht reviewed (commented):
LGTM (non-binding)

hyjorc1 reviewed (approved):
r=me, if bots are happy.

hyjorc1 commented on Source/WebCore/rendering/BidiRun.h:
```diff
@@ -33,6 +34,8 @@ class LegacyInlineBox;
 
 struct BidiRun : BidiCharacterRun {
+    WTF_MAKE_STRUCT_TZONE_ALLOCATED(BidiRun);
+public:
```
I think this `public` is also not needed right? Since `WTF_MAKE_STRUCT_TZONE_ALLOCATED` has the definition:
```
#define MAKE_STRUCT_BTZONE_MALLOCED(isoType, heapType, exportMacro) \
    MAKE_BTZONE_MALLOCED_COMMON(isoType, heapType, exportMacro) \
    using __makeTZoneMallocedMacroSemicolonifier BUNUSED_TYPE_ALIAS = int
```

msaboff committed:
[TZone] Add TZone annotation macros for structs
https://bugs.webkit.org/show_bug.cgi?id=281514
rdar://137985417

Reviewed by Yijia Huang.

Split the existing MAKE_BTZONE_MALLOCED macro into MAKE_BTZONE_MALLOCED_COMMON, that doesn't have access specifiers,
and MAKE_BTZONE_MALLOCED, which invokes MAKE_BTZONE_MALLOCED_COMMON with a public specifier before and a
private specifier after.  Created a new MAKE_STRUCT_BTZONE_MALLOCED that just invokes MAKE_BTZONE_MALLOCED_COMMON.
Both the class and struct macros include a dummy using statement to allow them to be followed by a semicolon.
Made similar changes to MAKE_BTZONE_MALLOCED_INLINE.

Added WTF_MAKE_STRUCT_TZONE_ALLOCATED and WTF_MAKE_STRUCT_TZONE_ALLOCATED_INLINE which call the new MAKE_BTZONE_* macros.

Used both of the new WTF_MAKE_STRUCT_TZONE_ALLOCATED* macros in a few files, replacing WTF_MAKE_STRUCT_FAST_ALLOCATED in
some places and added WTF_MAKE_STRUCT_TZONE_ALLOCATED in a few places where there wasn't a WTF_MAKE_STRUCT_FAST_ALLOCATED.

Note that the nested struct RemoteObjectInvocation::ReplyInfo was previously annotated using the "class" annotation macro.
This worked because RemoteObjectInvocation::ReplyInfo had an existing public: specifier.  For consistency,
changed RemoteObjectInvocation::ReplyInfo to use WTF_MAKE_STRUCT_TZONE_ALLOCATED instead and removed the public: specifier.

* Source/JavaScriptCore/bytecode/LazyValueProfile.h:
* Source/JavaScriptCore/bytecode/UnlinkedFunctionExecutable.cpp:
* Source/JavaScriptCore/bytecode/UnlinkedFunctionExecutable.h:
* Source/JavaScriptCore/dfg/DFGOSRExit.cpp:
* Source/JavaScriptCore/dfg/DFGOSRExit.h:
* Source/JavaScriptCore/inspector/remote/socket/RemoteInspectorSocketEndpoint.cpp:
* Source/JavaScriptCore/inspector/remote/socket/RemoteInspectorSocketEndpoint.h:
* Source/JavaScriptCore/jit/JITMathIC.h:
* Source/JavaScriptCore/jit/JITTZoneImpls.cpp:
* Source/JavaScriptCore/parser/VariableEnvironment.cpp:
* Source/JavaScriptCore/parser/VariableEnvironment.h:
* Source/WTF/wtf/TZoneMalloc.h:
* Source/WTF/wtf/TZoneMallocInlines.h:
* Source/WebCore/Modules/airplay/WebMediaSessionManager.cpp:
* Source/WebCore/PAL/pal/crypto/commoncrypto/CryptoDigestCommonCrypto.cpp:
* Source/WebCore/rendering/BidiRun.cpp:
* Source/WebCore/rendering/BidiRun.h:
* Source/WebKit/Shared/API/Cocoa/RemoteObjectInvocation.h:
* Source/bmalloc/bmalloc/TZoneHeap.h:
* Source/bmalloc/bmalloc/TZoneHeapInlines.h:

Canonical link: https://commits.webkit.org/285280@main
